### PR TITLE
Fix global sound event gamerule not being respected

### LINKED ISSUE #8726: globalSoundEvents game-rule not working correctly
### Expected behavior

With the `globalSoundEvents` game-rule (added in 1.19.3) enabled, sound events like wither summoning shouldn't be heard across the whole server.

### Observed/Actual behavior

The wither summoning sound can be heard across the whole server.

### Steps/models to reproduce

1. Set globalSoundEvents to false
2. Place one account in the overworld
3. Place the second account in the nether
4. Summon a wither in one world
5. Hear the sound with the other account

### Plugin and Datapack List

None

### Paper version

[01:03:17 INFO]: This server is running Paper version git-Paper-359 (MC: 1.19.3) (Implementing API version 1.19.3-R0.1-SNAPSHOT) (Git: 3331501)
You are running the latest version

### Other

_No response_

### LINKED ISSUE #9182: Ender Dragon death noise plays everywhere
### Expected behavior

When the Ender Dragon is killed, only those on the main End island should hear the death noise.

### Observed/Actual behavior

When the Ender Dragon is killed, the (rather loud) death noise is played to everyone in any dimension, which is rather unsettling to those who are doing other things in the Overworld and did not know someone was fighting an Ender Dragon.

### Steps/models to reproduce

1) have someone in the End attacking an Ender Dragon
2) have someone in another dimension (e.g. the Overworld)
3) The person in the End kills the Ender Dragon
4) Every single person in the server hears the death noise

### Plugin and Datapack List

None
image

### Paper version

This server is running Paper version git-Paper-524 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 9846d0d)
You are running the latest version
image

### Other

This is consistent across two different Paper servers running different versions.

### PR DESCRIPTION
**Pull request by Machine-Maker**
Changes handling of the global sounds. Previously the global sound gamerule wasn't used correctly resulting in unexpected behavior (fixes #8726, fixes #9182 ). This changes the `spigot.yml` behavior of setting the sound radius for the 3 events to 0. Now, if set to 0, the default radius is used (64). The behavior is now controlled by the gamerule, only using the configuration for the actual radius, not whether to enable it or not.

---
Download the paperclip jar for this pull request: [paper-8727.zip](https://nightly.link/PaperMC/Paper/actions/artifacts/1137137340.zip)

### CODE REVIEW

**Review by aerulion — APPROVED** [review]

File: patches/server/0952-Fix-global-sound-event-gamerule-not-being-respected.patch [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                             double relativeX = player.getX() + (deltaX / deltaLength) * viewDistance;
+diff --git a/src/main/java/net/minecraft/world/item/EnderEyeItem.java b/src/main/java/net/minecraft/world/item/EnderEyeItem.java
+index f4fc8ff3981555e4b560289248a9b02afc04b42f..519bad9a16eab8e343f9340f7604115ce9c09bae 100644
+--- a/src/main/java/net/minecraft/world/item/EnderEyeItem.java
++++ b/src/main/java/net/minecraft/world/item/EnderEyeItem.java
+@@ -16,6 +16,7 @@ import net.minecraft.world.entity.player.Player;
+ import net.minecraft.world.entity.projectile.EyeOfEnder;
+ import net.minecraft.world.item.context.UseOnContext;
+ import net.minecraft.world.level.ClipContext;
++import net.minecraft.world.level.GameRules;
```
Comment (NoahvdAa) [inline patches/server/0952-Fix-global-sound-event-gamerule-not-being-respected.patch]: import

**Review by NoahvdAa — COMMENTED** [review]
I feel like this should be merged into [this patch](https://github.com/PaperMC/Paper/blob/1975fbe3686be68d80bcfff4a4478afe45a3df10/patches/server/0101-Only-send-global-sounds-to-same-world-if-limiting-ra.patch)

**Comment by kennytv** [comment]
I feel like this should go into 0101 and be called "Fix CB global sound handling" now

File: patches/server/0952-Fix-global-sound-event-gamerule-not-being-respected.patch [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                     BlockPos soundPos = blockposition1.offset(1, 0, 1);
+-                    for (ServerPlayer player : world.spigotConfig.endPortalSoundRadius > 0 ? ((ServerLevel) world).players() : world.getServer().getPlayerList().players) { // Paper
++                    for (ServerPlayer player : !world.getGameRules().getBoolean(GameRules.RULE_GLOBAL_SOUND_EVENTS) ? ((ServerLevel) world).players() : world.getServer().getPlayerList().players) { // Paper - respect global sound events gamerule
+                         double deltaX = soundPos.getX() - player.getX();
+                         double deltaZ = soundPos.getZ() - player.getZ();
+                         double distanceSquared = deltaX * deltaX + deltaZ * deltaZ;
+-                        if (world.spigotConfig.endPortalSoundRadius > 0 && distanceSquared > world.spigotConfig.endPortalSoundRadius * world.spigotConfig.endPortalSoundRadius) continue; // Spigot
++                        final double soundRadiusSquared = world.spigotConfig.endPortalSoundRadius == 0 ? 64 * 64 : world.spigotConfig.endPortalSoundRadius * world.spigotConfig.witherSpawnSoundRadius; // Paper - respect global sound events gamerule
```
Comment (Lulu13022002) [inline patches/server/0952-Fix-global-sound-event-gamerule-not-being-respected.patch]: Should be endPortalSoundRadius * endPortalSoundRadius (actually it's endPortalSoundRadius * witherSpawnSoundRadius)

**Comment by Machine-Maker** [comment]
Addressed requested changes

**Comment by kennytv** [comment]
Should the config really just be ignored if the gamerule isn't set to false? Imo having a config value > 0 definitely means you want this to be used and the sounds to not be global

**Comment by Machine-Maker** [comment]
> Should the config really just be ignored if the gamerule isn't set to false? Imo having a config value > 0 definitely means you want this to be used and the sounds to not be global

maybe we add a migration then which enables initially sets the gamerule if the config is above 0. But we should defer to the “vanilla setting” first. Then look at the spigot config.

**Comment by Netherwhal** [comment]
What is the current way forward? Ideally we should prefer vanilla behavior?

**Comment by Machine-Maker** [comment]
Rebased for 1.20.1

**Review by Warriorrrr — APPROVED** [review]

**Review by Owen1212055 — APPROVED** [review]

**Review by lynxplay — APPROVED** [review]

**Pull request merged**